FreeRTOS_TCP_IP.c

```diff
@@ -418,6 +418,47 @@
     }
 /*-----------------------------------------------------------*/
 
+/**
+ * @brief Check whether the address is unicast of multicast/broadcast.
+ *
+ * @param[in] pxAddress: The structure containing the IP-address to be
+ *            checked (in 32-bit format).
+ *
+ * @return pdTRUE if the IP address is either multicast or broadcast. pdFALSE
+ *         otherwise.
+ *
+ * @note The structure is used instead of a 32-bit IP is to make this function
+ *       similar in signature for IPv6.
+ */
+    BaseType_t xIsUnicastAddress( struct freertos_sockaddr * pxAddress )
+    {
+        BaseType_t xResult = pdTRUE;
+        uint32_t ulIPAddress = pxAddress->sin_addr;
+
+        if( ( FreeRTOS_ntohl( ulIPAddress ) & 0xffU ) == 0xffU )
+        {
+            /* This is a broadcast address x.x.x.255. */
+            xResult = pdFALSE;
+        }
+        else if( xIsIPv4Multicast( ulIPAddress ) != pdFALSE )
+        {
+            /* This is a multicast address. */
+            xResult = pdFALSE;
+        }
+        else if( ulIPAddress == 0U )
+        {
+            /* This is a non-standard broadcast address. */
```

Comment:
You can also use the value `FREERTOS_INADDR_ANY`, instead of `0U`.
Comment is not correct, it is a "an invalid, unknown or non-applicable target".
Quoted text borrowed from Wikipedia: https://en.wikipedia.org/wiki/0.0.0.0
( that is why I wanted `inet_pton()` to clear the IP-address ).